presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -166,8 +166,8 @@ public boolean isNull()
         @Override
         public double getDouble()
         {
-            if (value instanceof Double || value instanceof Float) {
-                return ((Number) value).doubleValue();
```

Comment:
Good catch. We should really avoid using `Number` interface.
It's not the first time it's causing correctness problems.